pkg/policies/engine/rego/rego.go

```diff
@@ -145,7 +151,51 @@ func (r *Rego) Verify(ctx context.Context, policy *engine.Policy, input []byte,
 		}
 	}
 
-	return violations, nil
+	return &engine.EvaluationResult{
+		Violations: violations,
+		Skipped:    false, // best effort
+		Message:    "",
+	}, nil
+}
+
+func parseResultRule(res rego.ResultSet, policy *engine.Policy) (*engine.EvaluationResult, error) {
+	result := &engine.EvaluationResult{Violations: make([]*engine.PolicyViolation, 0)}
+	for _, exp := range res {
+		for _, val := range exp.Expressions {
+			ruleResult, ok := val.Value.(map[string]any)
+			if !ok {
+				return nil, fmt.Errorf("failed to evaluate policy evaluation result: %s", val.Text)
```

Comment:
Shall we create a custom error when the expected fields are not present in the script?